tests/cases/compiler/interfaceClassMerging2.ts

```diff
@@ -0,0 +1,35 @@
+interface Foo {
+    interfaceFooMethod(): this;
+    interfaceFooProperty: this;
+}
+
+class Foo {
+    classFooProperty: this;
+
+    classFooMethod(): this {
+        return this;
+    }
+}
+
+
+interface Bar {
+    interfaceBarMethod(): this;
+    interfaceBarProperty: this;
+}
+
+class Bar extends Foo {
+    classBarProperty: this;
+
+    classBarMethod(): this {
+        return this;
+    }
+}
+
+
+var bar = new Bar();
+bar.classFooMethod().interfaceFooMethod().classBarMethod().classFooMethod();

```

Comment:
I kind of expected the last method to be `interfaceBarMethod();` since all the others are used and `classFooMethod()` appears twice.